lerobot/common/robot_devices/robots/manipulator.py

```diff
@@ -391,48 +394,151 @@ def connect(self):
-                calibration[f"leader_{name}"][motor_name] = (homing_offset[idx], drive_mode[idx])
+            # Use 'position control current based' for follower gripper to be limited by the limit of the current.
+            # It can grasp an object without forcing too much even tho,
+            # it's goal position is a complete grasp (both gripper fingers are ordered to join and reach a touch).
+            # 5 corresponds to Current Controlled Position on Aloha gripper follower "xm430-w350"
+            self.follower_arms[name].write("Operating_Mode", 5, "gripper")
 
-        return calibration
+            # Note: We can't enable torque on the leader gripper since "xc430-w150" doesn't have
+            # a Current Controlled Position mode.
+
+        if self.config.gripper_open_degree is not None:
+            warnings.warn(
+                f"`gripper_open_degree` is set to {self.config.gripper_open_degree}, but None is expected for Aloha instead",
+                stacklevel=1,
+            )
 
     def teleop_step(
         self, record_data=False
     ) -> None | tuple[dict[str, torch.Tensor], dict[str, torch.Tensor]]:
         if not self.is_connected:
             raise RobotDeviceNotConnectedError(
-                "KochRobot is not connected. You need to run `robot.connect()`."
+                "ManipulatorRobot is not connected. You need to run `robot.connect()`."
             )
 
         # Prepare to assign the position of the leader to the follower
         leader_pos = {}
         for name in self.leader_arms:
             before_lread_t = time.perf_counter()
             leader_pos[name] = self.leader_arms[name].read("Present_Position")
+            leader_pos[name] = torch.from_numpy(leader_pos[name])
             self.logs[f"read_leader_{name}_pos_dt_s"] = time.perf_counter() - before_lread_t
 
+        # Send goal position to the follower
         follower_goal_pos = {}
-        for name in self.leader_arms:
-            follower_goal_pos[name] = leader_pos[name]
-
-        # Send action
         for name in self.follower_arms:
             before_fwrite_t = time.perf_counter()
-            self.send_action(torch.tensor(follower_goal_pos[name]), [name])
+            goal_pos = leader_pos[name]
+
+            # Cap goal position when too far away from present position.
+            # Slower fps expected due to reading from the follower.
+            if self.config.max_relative_target is not None:
+                present_pos = self.follower_arms[name].read("Present_Position")
+                present_pos = torch.from_numpy(present_pos)
+                goal_pos = ensure_safe_goal_position(goal_pos, present_pos, self.config.max_relative_target)
+
+            # Used when record_data=True
+            follower_goal_pos[name] = goal_pos
+
+            goal_pos = goal_pos.numpy().astype(np.int32)
+            self.follower_arms[name].write("Goal_Position", goal_pos)
```

Comment:
Redesign of `max_relative_target` logic
cc @alexander-soare

Reply:
@Cadene we have a `send_action` method so I think it makes more sense to use it than to duplicate code here, especially for something as critical as this. My initial intent with the `follower_names` arg was to make your loop here work. But really, I think the bes solution is to grap all of the goal positions in one array and when the loop is finished use `send_actions`.

Reply:
I am not sure.

1. The logic of `capture_observation` and `teleop_step(record_data=True)` consists in applying steps to capture and tranform data from the arms so that **at the end** tensors of state and action are returned. Then the logic of `send_action` consists in applying steps to un-transform action tensor into actionnable data for the arms. `send_action` is not supposed to be used inside `telop_step`. There is something wrong in term of "interface".

2. Providing `follower_arms` as argument of `send_action` feels wrong to me, in addition to the weird syntax which might reveal a design issue:
```python
self.send_action(torch.tensor(follower_goal_pos[name]), [name])
```

Reply:
Yes let's drop 2 as I'm happy to remove the `follower_arms` arg.

Sorry but I'm not sure what you mean with 1. I can't see how this means that `send_action` shouldn't be used in teleop step. I'll trust that you have the right idea though. But please consider how you might reuse the action sending logic somehow as IMO it's hard to maintain / too easy to break. Maybe a private method that contains just the shared logic...

Unblocking.